core-bundle/src/Repository/CronJobRepository.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Repository;
+
+use Contao\CoreBundle\Entity\CronJob;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\Common\Persistence\ManagerRegistry;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @internal
+ *
+ * @method object|null findOneByName(string $name)
+ */
+class CronJobRepository extends ServiceEntityRepository
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, CronJob::class);
+
+        if (($connection = $registry->getConnection()) instanceof Connection) {
```

Comment:
You are checking for a particular instance here but you never handle the case that the check is not true. If `$connection` is not an instance of `Connection`, the other methods of the class will fail.

Reply:
To be honest, this class is more or less a copy & paste of the existing `RememberMeRepository`. The same happens there too:

https://github.com/contao/contao/blob/133696aa02ef2dfb4005937e25b8c8427de885ec/core-bundle/src/Repository/RememberMeRepository.php#L30-L37

I don't really know what the purpose of this check is. If the `ManagerRegistry` has no valid database connection, what difference does this make for the repository?

Reply:
This needs to be fixed in the `RememberMeRepository` as well. You can leave the class as is and add a ticket as reminder.